nix/nix/sources.json

```diff
@@ -36,10 +36,10 @@
         "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz"
     },
     "opam-repository": {
-        "branch": "master",
-        "ref": "master",
+        "branch": "",
```

Comment:
Hmmmmm

Reply:
That doesn't look right 😞

Reply:
But the new command for finding `$opam_repository_branch` really does return an empty string :coolstory:

Reply:
For some reason it lists refs as `refs/heads/master` instead of any `refs/remotes/origin`, I assume that's why various incorrect filtering happens

Reply:
Aight, I think I made it more robust against these changing `refname`s by assuming the branches in `tezos/opam-repository` never contain slashes. @rvem What do you think?

Reply:
possibly that was a problem with my suggestions because I checked it with a non-`bare` cloned repo.
In fact it seems to be still working here as well

Reply:
_I left [a suggestion below](https://github.com/serokell/tezos-packaging/pull/458#discussion_r866188858) which should work also with slashes in the branch names_